lib/runtime-core/src/state.rs

```diff
@@ -86,8 +109,84 @@ pub enum MachineValue {
     TwoHalves(Box<(MachineValue, MachineValue)>), // 32-bit values. TODO: optimize: add another type for inner "half" value to avoid boxing?
 }
 
+impl BorshSerialize for MachineValue {
+    fn serialize<W: std::io::Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        match self {
+            MachineValue::Undefined => writer.write_all(&[0u8])?,
+            MachineValue::Vmctx => writer.write_all(&[1u8])?,
+            MachineValue::VmctxDeref(v) => {
+                writer.write(&[2u8])?;
+                BorshSerialize::serialize(&v, writer)?;
+            }
+            MachineValue::PreserveRegister(r) => {
+                writer.write(&[3u8])?;
+                BorshSerialize::serialize(&r, writer)?;
+            }
+            MachineValue::CopyStackBPRelative(i) => {
+                writer.write(&[4u8])?;
+                BorshSerialize::serialize(&i, writer)?;
+            }
+            MachineValue::ExplicitShadow => writer.write_all(&(5 as u8).to_le_bytes())?,
```

Comment:
This should be just `writer.write_all(&[5u8])?` for consistency

Reply:
Sorry i think this one is missed, to_le_bytes() does nothing to u8 so correctness is fine, but in a refactor i missed changing this case